pkg/fleetautoscalers/fleetautoscalers_test.go

```diff
@@ -1162,6 +1189,7 @@ func TestApplyCounterPolicy(t *testing.T) {
 					Capacity: 7}
 			}),
 			featureFlags: string(utilruntime.FeatureCountsAndLists) + "=true",
+			// TODO: Is this what we want (replica = 0 if MaxCapcity < per Counter Capacity)? If buffer = 1 then must be at least replica = 1? Or Max Capacity takes precedence?
```

Comment:
Is this what we want (replica = 0 if MaxCapcity < per Counter Capacity)? If buffer = 1 then must be at least replica = 1? Or Max Capacity takes precedence?

Test case: https://github.com/googleforgames/agones/blob/2003e674cd3c8a3fceac5ae2f2fb5c3a5f5fc48f/pkg/fleetautoscalers/fleetautoscalers_test.go#L1149-L1187

Reply:
Yeah, a requested buffer shouldn't ever be less than 0 (for regular buffer it's a min value of 1), so there must always be at least one replica.

Scaling to zero is way too hard 😃 since we can't control game server spinup time.

Reply:
The requested buffer here is 1, but it's the MaxCapacity that's taking precedence, and since MaxCapacity < per Counter Capacity it scales to 0. 

Just to clarify, in general, you're saying we should never scale to 0?

Reply:
There has to be some number of gameserver(s) to have some allocatable capacity to go towards the buffer - so there should always be at least one, as otherwise that would go against what the fleet autoscaler is requesting.

Reply:
Sounds good -- I'll add in a check to make sure we never return less than 1 desired replica.